### PR TITLE
[Static reference] LPS-182671 Remove static references in class MBDiscussionPermission | part 2

### PR DESCRIPTION
**Pull request by W2jx**
resend

Previous: https://github.com/liferay-core-infra/liferay-portal/pull/5070

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:lima

**Comment by W2jx** [comment]
ci:test:comment

**Comment by W2jx** [comment]
ci:test:message-boards

**Comment by W2jx** [comment]
ci:test:questions

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:lima

**Comment by W2jx** [comment]
ci:test:comment

**Comment by W2jx** [comment]
ci:test:message-boards

**Comment by W2jx** [comment]
ci:test:questions

File: portal-kernel/src/com/liferay/portal/kernel/util/SubscriptionSender.java [hunk 1]
```diff
@@ -660,12 +657,20 @@ protected Boolean hasSubscribePermission(
...
-				permissionChecker, groupId, subscription.getClassName(),
+				modelResourcePermission, permissionChecker, groupId,
 				subscription.getClassPK(), ActionKeys.SUBSCRIBE);
 		}
 
-		return Boolean.TRUE;
+		return true;
 	}
 
 	protected void notifyPersistedSubscriber(
```
Comment (yuhai) [inline portal-kernel/src/com/liferay/portal/kernel/util/SubscriptionSender.java:694]: @W2jx, we may simplify the logic as the following:
```
if (resourceAction == null) {
  return true;
}

ModelResourcePermission<?> modelResourcePermission =
	ModelResourcePermissionRegistryUtil.getModelResourcePermission(
			subscription.getClassName());
...

return BaseModelPermissionCheckerUtil.containsBaseModelPermission();
```

**Comment by yuhai** [comment]
ci:stop

**Comment by Kvale1733** [comment]
Adding ignore label for now since change was requested. Please remove once it's made

File: portal-kernel/src/com/liferay/portal/kernel/util/SubscriptionSender.java (hunk 1 above)
Comment (W2jx) [reply portal-kernel/src/com/liferay/portal/kernel/util/SubscriptionSender.java:694]: OK

**Review by yuhai — APPROVED** [review]
@dantewang, the pr is good to me.

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:lima

**Comment by W2jx** [comment]
ci:test:comment

**Comment by W2jx** [comment]
ci:test:message-boards

**Comment by W2jx** [comment]
ci:test:questions

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by W2jx** [comment]
no conflicts(7/28 9:24)

**Comment by tinatian** [comment]
@W2jx This is a big pull, is it possible to separate it into smaller ones?

Thanks.
Tina

**Pull request closed without merging**